### PR TITLE
Fix #133: Allow some special chars in table name

### LINKED ISSUE #133: Allow to use dot in middle of table name in AR command
For example, table with scheme in PostgreSQL: `booking.room`.

Need to fix `Regex` validation rule:

https://github.com/yiisoft/yii-gii/blob/ca67155160516f6d6710a249921f36758f916291/src/Generator/ActiveRecord/Command.php#L26-L31

### PR DESCRIPTION
**Pull request by ahmadasjad**
Fixes #133 

Table name regex correction
Exception handling for syntax error when passing a wrong table name

Regex test: https://regex101.com/r/QmSWaW/1

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Generator/ActiveRecord/Command.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ public function __construct(
         private readonly string $namespace = 'App\\Model',
         #[Required]
         #[Regex(
-            pattern: '/^[a-z_][a-z0-9_]*$/i',
+            pattern: '/^[\w-]+(?:\.[\w-]+)?$/i',
```
Comment (vjik) [inline src/Generator/ActiveRecord/Command.php:27]:
```suggestion
            pattern: '/^[\w-]+\.?[\w-]*$/i',
```

Comment (vjik) [inline src/Generator/ActiveRecord/Command.php:27]: Can you add tests?

File: src/Validator/TableExistsHandler.php [hunk 2]
```diff
@@ -28,7 +28,11 @@ public function validate(mixed $value, object $rule, ValidationContext $context)
 
         $result = new Result();
 
-        $tableSchema = $this->connection->getTableSchema($value);
+        try {
+            $tableSchema = $this->connection->getTableSchema($value);
+        } catch (\Yiisoft\Db\Exception\Exception) {
+            $tableSchema = null;
```
Comment (vjik) [inline src/Validator/TableExistsHandler.php:34]: I think, better add validation error with exception message instead of suppressing.

**Review by vjik — CHANGES_REQUESTED** [review]

File: src/Validator/TableExistsHandler.php (hunk 2 above)
Comment (ahmadasjad) [reply src/Validator/TableExistsHandler.php]: added

File: src/Generator/ActiveRecord/Command.php (hunk 1 above)
Comment (ahmadasjad) [reply src/Generator/ActiveRecord/Command.php]: I didn't add it as the test environment isn't set up yet in the repo.
An outside test can be seen here: https://regex101.com/r/aLjS42/1

Comment (xepozz) [reply src/Generator/ActiveRecord/Command.php]: Actually there are some bugs: https://regex101.com/r/KX8vqz/1

Comment (xepozz) [reply src/Generator/ActiveRecord/Command.php]: Maybe kind of this https://regex101.com/r/B8gCXv/1

Comment (vjik) [reply src/Generator/ActiveRecord/Command.php]: Also may be several dots. For example, in MSSQL: `database_name.schema_name.table_name`.

The table name specification is needed.

Comment (vjik) [reply src/Generator/ActiveRecord/Command.php]: I suggest to check only symbols, for example, `/^[\w-"`]+$/`. 
And do a more detailed check later.

Comment (ahmadasjad) [reply src/Generator/ActiveRecord/Command.php]: > I suggest to check only symbols, for example, `/^[\w-"`]+$/`. And do a more detailed check later.

I'm also of the same opinion

**Review by samdark — COMMENTED** [review]
Code itself looks good :+1:

File: src/Generator/ActiveRecord/Command.php (hunk 1 above)
Comment (ahmadasjad) [reply src/Generator/ActiveRecord/Command.php]: Let me know which one I should use.
```
^(["'`]?)([\w][\w-]*)\1(?:\.((?1))(?2)\3)*$
```
https://regex101.com/r/3zbKmI/1

OR
```
^[\w\-."`]+$
```
https://regex101.com/r/3b3NYH/1

Comment (xepozz) [reply src/Generator/ActiveRecord/Command.php]: afaik `-` is not permitted to use, isn't it?

Comment (ahmadasjad) [reply src/Generator/ActiveRecord/Command.php]: https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
tested in MySql, it's permitted

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**